pkg/tuple/validation.go

```diff
@@ -18,6 +18,11 @@ func (i *InvalidTupleError) Error() string {
 	return fmt.Sprintf("Invalid tuple '%s'. Reason: %s", i.TupleKey, i.Reason)
 }
 
+func (i *InvalidTupleError) Is(target error) bool {
```

Comment:
Are these `Is` functions used anywhere? Are they necessary? Is this just a convenience over using `errors.As`?

Reply:
They are used in various places in the tests `require.ErrorIs(...)` uses them.

Reply:
In general the standard Go error spec outlines the `As` and `Is` functions. These implement the `Is` spec for the custom go errors we use all over.

Reply:
Is it okay that we are just checking the type? We don't need to check that the tuple and the reason are equal?

Reply:
I chose to do it like it is for the scope of this PR just to check for type equality which allows me to `require.ErrorIs(..)` in the tests and make sure the underlying error is of the specific custom error types. If we wanted our custom error check for `errors.Is(..)` to include the strings of the error itself then we'd want to do that, yeah.

In the context of where you use `errors.Is` most, you are doing so to get typed error checking semantics on wrapped custom errors. So I think implementing the `Is` method for our custom errors by just checking the type is a great way to do that.

Reply:
Cool cool.